File: atlaskv-java-sdk/src/main/java/com/atlaskv/sdk/models/WatchEvent.java
package com.atlaskv.sdk.models;

/**
 * Represents a key-value change event pushed over a watch stream.
 *
 * @param type  operation type (e.g. PUT, DELETE, EXPIRE)
 * @param key   mutated key
 * @param value value of the key (null if deleted or expired)
 */
public record WatchEvent(
        String type,
        String key,
        String value
) {}

File: atlaskv-ts-sdk/src/models/WatchEvent.ts
/**
 * Represents a real-time key-value change event pushed over a watch stream.
 */
export interface WatchEvent {
  /** The mutation operation type (e.g., "PUT", "DELETE", "EXPIRE"). */
  type: string;
  /** The mutated key. */
  key: string;
  /** The new value of the key (null if deleted or expired). */
  value: string | null;
}
